src/handlers/handleIngressAPI.ts

```diff
@@ -46,26 +49,61 @@ function createResponseWithFirstPartyCookies(request: Request, response: Respons
   })
 }
 
-async function makeIngressAPIRequest(request: Request, env: WorkerEnv) {
-  const oldURL = new URL(request.url)
-  const endpoint = getVisitorIdEndpoint(oldURL.searchParams)
+function createRequestURL(receivedRequestURL: string, routeMatches: RegExpMatchArray | undefined) {
+  const routeSuffix = routeMatches ? routeMatches[1] : undefined
+  const oldURL = new URL(receivedRequestURL)
+  const endpoint = getVisitorIdEndpoint(oldURL.searchParams, routeSuffix)
   const newURL = new URL(endpoint)
   copySearchParams(oldURL, newURL)
-  addTrafficMonitoringSearchParamsForVisitorIdRequest(newURL)
 
-  let headers = new Headers(request.headers)
-  addProxyIntegrationHeaders(headers, env)
+  return newURL
+}
+
+async function makeIngressRequest(
+  receivedRequest: Request,
+  env: WorkerEnv,
+  routeMatches: RegExpMatchArray | undefined,
+) {
+  const requestURL = createRequestURL(receivedRequest.url, routeMatches)
+  addTrafficMonitoringSearchParamsForVisitorIdRequest(requestURL)
+  let headers = new Headers(receivedRequest.headers)
   headers = filterCookies(headers, (key) => key === '_iidt')
+  addProxyIntegrationHeaders(headers, env)
+  const body = await (receivedRequest.headers.get('Content-Type') ? receivedRequest.blob() : Promise.resolve(null))
+  console.log(`sending ingress request to ${requestURL}...`)
+  const request = new Request(requestURL, new Request(receivedRequest, { headers, body }))
+
+  return fetch(request).then((response) => createResponseWithFirstPartyCookies(receivedRequest, response))
+}
 
-  console.log(`sending ingress api to ${newURL}...`)
-  const body = await (request.headers.get('Content-Type') ? request.blob() : Promise.resolve(null))
-  const newRequest = new Request(newURL.toString(), new Request(request, { headers, body }))
-  return fetch(newRequest).then((response) => createResponseWithFirstPartyCookies(request, response))
+function makeCacheRequest(receivedRequest: Request, routeMatches: RegExpMatchArray | undefined) {
```

Comment:
We probably should name this function as’makeCacheableRequest’. makeCacheRequest looks like we are calling a service that puts something in cache.

Reply:
We are making a request to the cache endpoint. I can rename it as `makeCacheEndpointRequest` how does that sound? @Orkuncakilkaya

Reply:
Makes more sense! 👍